frontends/mit-open/src/pages/FieldPage/FieldPage.test.tsx

```diff
@@ -35,11 +34,7 @@ const setupApis = (
     field,
   )
   setMockResponse.get(
-    urls.learningResources.featured({ limit: 12, platform: ["ocw"] }),
-    factories.learningResources.resources({ count: 0 }),
-  )
-  setMockResponse.get(
-    urls.learningResources.featured({ limit: 12 }),
+    expect.stringContaining(urls.learningResources.featured()),
```

Comment:
The flakiness was being caused by the fact that the API url on line 38 wasn't always correct—it depends on the `search_filter`. In some tests it was *always* wrong, but the test would pass before any api call would be attempted, meaning we never got an error. 

To fix this, I made the mocking more flexible... now it just needs to contain `urls.learningResources.featured()`. Since this is much more flexible, I added an assertion to one of the tests that the correct API call is actually made.